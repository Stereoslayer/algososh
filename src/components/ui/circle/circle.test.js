import renderer from "react-test-renderer";
import {Circle} from "./circle";
import {render, screen, within} from "@testing-library/react";
import {ElementStates} from "../../../types/element-states";


describe('Circle test', () => {
    it('Circle without letter test', () => {
        const circle = renderer.create(<Circle/>).toJSON();
        expect(circle).toMatchSnapshot();
    })
    it('Circle with letter test', () => {
        const props = {
            letter: '0'
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('circle')).toHaveTextContent('0');
        expect(circle).toMatchSnapshot();
    })
    it('Circle with head test', () => {
        const props = {
            head: 'someValue'
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('head')).toHaveTextContent('someValue');
        expect(circle).toMatchSnapshot();
    })
    it('Circle with react-element in head test', () => {
        const props = {
            head: <Circle isSmall={true}/>
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        within(screen.getByTestId('circle')).getByTestId('circleSmall');
        expect(circle).toMatchSnapshot();
    })
    it('Circle with tail test', () => {
        const props = {
            tail: 'someValue'
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('tail')).toHaveTextContent('someValue');
        expect(circle).toMatchSnapshot();
    })
    it('Circle with react-element in tail test', () => {
        const props = {
            tail: <Circle isSmall={true}/>
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        within(screen.getByTestId('circle')).getByTestId('circleSmall');
        expect(circle).toMatchSnapshot();
    })
    it('Circle with index test', () => {
        const props = {
            index: 'someValue'
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('index')).toHaveTextContent('someValue');
        expect(circle).toMatchSnapshot();
    })
    it('Circle with prop isSmall test', () => {
        const props = {
            isSmall: true
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('isSmall')).toBeTruthy();
        expect(circle).toMatchSnapshot();
    })
    it('Circle default color test', () => {
        const props = {
            state: ElementStates.Default
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('colouredCircle')).toHaveClass('default');
        expect(circle).toMatchSnapshot();
    })
    it('Circle default color test', () => {
        const props = {
            state: ElementStates.Changing
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('colouredCircle')).toHaveClass('changing');
        expect(circle).toMatchSnapshot();
    })
    it('Circle default color test', () => {
        const props = {
            state: ElementStates.Modified
        };
        const circle = renderer.create(<Circle {...props} />).toJSON();
        render(<Circle {...props} />);
        expect(screen.getByTestId('colouredCircle')).toHaveClass('modified');
        expect(circle).toMatchSnapshot();
    })
})