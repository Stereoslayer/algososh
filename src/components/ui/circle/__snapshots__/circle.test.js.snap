// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle test Circle default color test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle default color test 2`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle default color test 3`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle with head test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    someValue
  </div>
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle with index test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    someValue
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle with letter test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      0
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle with prop isSmall test 1`] = `
<div
  className="content "
  data-testid="circleSmall"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle  small default"
    data-testid="isSmall"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle with react-element in head test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      className="content "
      data-testid="circleSmall"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle  small default"
        data-testid="isSmall"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle test Circle with react-element in tail test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      className="content "
      data-testid="circleSmall"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle  small default"
        data-testid="isSmall"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Circle test Circle with tail test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    someValue
  </div>
</div>
`;

exports[`Circle test Circle without letter test 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="colouredCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
